pkg/controller/replicaset/replica_set_utils.go

```diff
@@ -174,3 +177,118 @@ func filterOutCondition(conditions []extensions.ReplicaSetCondition, condType ex
 	}
 	return newConditions
 }
+
+// Obtain all RSs affected by pod creation event. The codes should be similar with addPod(),
+// but return RSs instead of enqueuing them to controller's workqueue.
+func getAffectedRSsByPodCreation(pod *v1.Pod, rsc *ReplicaSetController) []*extensions.ReplicaSet {
```

Comment:
The goal was to extract a useful helper that doesn't need any `ReplicaSetController`. However, I'm now thinking that may not be feasible at this time. The functions we depend on, like `getPodReplicaSets`, are tied too closely with the client code (`ReplicaSetLister`) to let us decouple the client calls from the decision making here. We would have to first refactor the client code itself, which is beyond the scope of our test refactor effort.

Let's focus on the other refactorings for now (like `manageReplicas`) and come back to this if we can think of a better way.